### PR TITLE
Use early-returns in TypeCombinator

### PR DESCRIPTION
**Pull request by staabm**
just a drive-by cleanup while studying the code. no functional changes.

motivation: improve readability by reducing the indentation

### CODE REVIEW

File: src/Type/TypeCombinator.php [hunk 1]
```diff
@@ -765,62 +765,62 @@ private static function optimizeConstantArrays(array $types): array
 	{
 		$constantArrayValuesCount = self::countConstantArrayValueTypes($types);
 
-		if ($constantArrayValuesCount > ConstantArrayTypeBuilder::ARRAY_COUNT_LIMIT) {
```
Comment (staabm) [inline src/Type/TypeCombinator.php:768]: inverted the condition, so we can de-indent the big logic block 1 level

File: src/Type/TypeCombinator.php [hunk 2]
```diff
@@ -765,62 +765,62 @@ private static function optimizeConstantArrays(array $types): array
...
-								}
-
-								return $traverse($type);
-							});
-							$valueTypes[$generalizedValueType->describe(VerbosityLevel::precise())] = $generalizedValueType;
-						}
+		$results = [];
+		foreach ($types as $type) {
+			$results[] = TypeTraverser::map($type, static function (Type $type, callable $traverse): Type {
+				if (!$type instanceof ConstantArrayType) {
```
Comment (staabm) [inline src/Type/TypeCombinator.php:775]: same as above: just added a early return so we can de-indent the code

**Comment by staabm** [comment]
see a shorted diff with https://github.com/phpstan/phpstan-src/pull/3197/files?diff=unified&w=1

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**